src/liballoc/collections/btree/search.rs

```diff
@@ -54,6 +59,12 @@ where
     Q: Ord,
     K: Borrow<Q>,
 {
+    // This function is defined over all borrow types (immutable, mutable, owned),
+    // and may be called on the shared root in each case.
+    // Crucially, we use `keys()` here, i.e., we work with immutable data.
+    // We do not need to make `keys_mut()` public and require support for the shared root.
```

Comment:
```suggestion
    // `keys_mut()` does not support the shared root, so we cannot use it.
```

Reply:
I don't understand why you want to say that about a non-public function elsewhere. We don't know if `keys_mut` even exists, at the moment of reading.

Reply:
It's not public but still part of the same data structure... so, someone who wanted to call it here would just slap a `pub` on it and not think much about it. At least, that's what people have been doing all over rustc all the time when they needed access to something private.